lib/modules/manager/gradle/artifacts.ts

```diff
@@ -147,7 +158,7 @@ export async function updateArtifacts({
       .map(quote)
       .join(' ')}`;
 
-    if (config.isLockFileMaintenance) {
+    if (config.isLockFileMaintenance || isGcvPropsFile(packageFileName)) {
```

Comment:
The behavior will be different for `gradle.lock` and `versions.lock`.

The `./gradlew --update-locks foo,bar` is triggered for normal `gradle.build/kts` build files, i.e. update gradle-locks in multiple locations.

However, if we are using GCV plugin, you will always have exactly one top-level lockfile, and it is updated with `--write-locks`. And the "changed" file will be `versions.props`, so that is what is happening here; we execute same logic as lockfileMaintenance if we have changes scheduled for `versions.props`. Thus I believe the logic is correct.

Reply:
In case of dependency updates, `updateArtifacts()` gets passed `versions.props` as `packageFileName`. In case of lock file maintenance, `packageFileName` is `build.gradle` or `build.gradle.kts`, otherwise it aborts (see lines 119-128).

`isGcvPropsFile` returns true in case it's passed *either* `versions.props` or the lock file in `packageFileName`. `packageFileName` should never be `versions.lock` though, so I'm questioning why this is included in this check - and sry for the confusion, it seems my statement above can be understood as the opposite.

Reply:
I re-worked the entire detection logic, so now there are separate `isGcvPropsFile` and `isGcvLockFile` methods etc. So leaving the check in line 161 as is will now mean that `--write-locks` is ran instead of `--update-locks` whenever we ahve our props file.